ablator/config/main.py

```diff
@@ -178,7 +178,19 @@ def annotations(self) -> dict[str, Annotation]:
         if hasattr(self, "__annotations__"):
             annotation_types = dict(self.__annotations__)
             # pylint: disable=no-member
```

Comment:
This should be removed?

Reply:
I think we can't remove `# pylint: disable=no-member`. 
For the follow code
```python
dataclass_types = {
                k: v.type
                for k, v in self.__dataclass_fields__.items()
                if k not in annotation_types
            }
```
`self.__dataclass_fields__` will trigger a  no-member error when run pylint.
Although the ConfigBase class is decorated with `@dataclass`, it doesn't define any fields. This might be the reason `__dataclass_fields__` is missing, causing the Pylint error.

Reply:
I think we can't remove `# pylint: disable=no-member`. 
For the follow code
```python
dataclass_types = {
                k: v.type
                for k, v in self.__dataclass_fields__.items()
                if k not in annotation_types
            }
```
`self.__dataclass_fields__` will trigger a  no-member error when run pylint.
Although the ConfigBase class is decorated with `@dataclass`, it doesn't define any fields. This might be the reason `__dataclass_fields__` is missing, causing the Pylint error.

Reply:
For make the code more stable.  Should we add `hasattr(self, "__dataclass_fields__")` before access to `__dataclass_fields__`

```python
 if hasattr(self, "__dataclass_fields__"):
                dataclass_types = {
                    k: v.type
                    for k, v in self.__dataclass_fields__.items()
                    if k not in annotation_types
                }
                annotation_types.update(dataclass_types)
```